pkg/curatedpackages/reader.go

```diff
@@ -61,14 +87,51 @@ func fetchPackages(ctx context.Context, versionsBundle releasev1.VersionsBundle,
 	}
 	images := make([]releasev1.Image, 0, len(bundle.Spec.Packages))
 	for _, p := range bundle.Spec.Packages {
-		pI := releasev1.Image{
+		pHC := releasev1.Image{
 			Name:        p.Name,
 			Description: p.Name,
 			OS:          ctrl.OS,
 			OSName:      ctrl.OSName,
-			URI:         fmt.Sprintf("%s/%s:%s", defaultRegistry, p.Source.Repository, p.Source.Versions[0].Name),
+			URI:         fmt.Sprintf("%s/%s:%s", GetRegistry(ctrl.URI), p.Source.Repository, p.Source.Versions[0].Name),
+		}
+		images = append(images, pHC)
+	}
+	return images, nil
+}
+
+func (r *PackageReader) fetchImagesFromBundle(ctx context.Context, versionsBundle releasev1.VersionsBundle, artifact string) ([]releasev1.Image, error) {
+	data, err := PullLatestBundle(ctx, artifact)
+	if err != nil {
+		return nil, err
+	}
+	bundle := &packagesv1.PackageBundle{}
+	err = yaml.Unmarshal(data, bundle)
+	if err != nil {
+		return nil, err
+	}
+
+	ctrl := versionsBundle.PackageController.Controller
+	images := make([]releasev1.Image, 0, len(bundle.Spec.Packages))
+
+	for _, p := range bundle.Spec.Packages {
+		// each package will have at least one version
+		for _, version := range p.Source.Versions[0].Images {
+			image := releasev1.Image{
+				Name:        version.Repository,
+				Description: version.Repository,
+				OS:          ctrl.OS,
+				OSName:      ctrl.OSName,
+				URI:         fmt.Sprintf("%s/%s@%s", getRegistry(ctrl.URI), version.Repository, version.Digest),
```

Comment:
Same question here about why we do this here in addition to what we do in the image mover

Reply:
I responded above but this doesn't perform any manipulation, it merely just extracts the images.